src/Console/Command/Init.php

```diff
@@ -44,42 +53,49 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             ],
             'Web application',
         );
+
         $templateDirectory = [
             'Web application' => 'http',
             'Event-driven function' => 'function',
         ][$choice];
 
-        $fs = new Filesystem;
-        $rootPath = dirname(__DIR__, 3) . "/template/$templateDirectory";
+        $this->rootPath = dirname(__DIR__, 3) . "/template/$templateDirectory";
+
+        self::createFile('index.php');
+        self::createFile('serverless.yml');
 
-        $io->writeln('Creating index.php');
-        $fs->copy("$rootPath/index.php", 'index.php');
+        $io->success('Project initialized and ready to test or deploy.');
 
-        $io->writeln('Creating serverless.yml');
+        return 0;
+    }
 
-        $template = file_get_contents("$rootPath/serverless.yml");
+    /**
+     * Creates files from the template directory and automatically adds
+     * them to git
+     */
+    private function createFile(string $file): void
+    {
+        $overwrite = true;
 
-        $template = str_replace('PHP_VERSION', PHP_MAJOR_VERSION . PHP_MINOR_VERSION, $template);
+        $this->io->writeln("Creating $file");
 
-        file_put_contents('serverless.yml', $template);
+        if (file_exists($file)) {
+            $overwrite = $this->io->confirm("A file named $file already exists, do you want to overwrite it?", true);
+        }
 
-        $filesToGitAdd = ['index.php', 'serverless.yml'];
+        if ($overwrite) {
+            $template = file_get_contents("$this->rootPath/$file");
+            $template = str_replace('PHP_VERSION', PHP_MAJOR_VERSION . PHP_MINOR_VERSION, $template);
+            file_put_contents($file, $template);
 
-        /*
-         * We check if this is a git repository to automatically add files to git.
-         */
-        if ((new Process(['git', 'rev-parse', '--is-inside-work-tree']))->run() === 0) {
-            foreach ($filesToGitAdd as $file) {
-                (new Process(['git', 'add', $file]))->run();
+            /*
+             * We check if this is a git repository to automatically add file to git.
+             */
+            if ((new Process(['git', 'rev-parse', '--is-inside-work-tree']))->run() === 0) {
+                    (new Process(['git', 'add', $file]))->run();
             }
-            $io->success([
-                'Project initialized and ready to test or deploy.',
-                'The files created were automatically added to git.',
-            ]);
-        } else {
-            $io->success('Project initialized and ready to test or deploy.');
-        }
 
-        return 0;
+            $this->io->success("$file successfully created and added to git automatically.");
```

Comment:
The message could be conditioned with the previous if-block to show "added to git..." only when the user is in a git repo.

Reply:
Do you mean something like this?

```php
$successMessage = "$file successfully created"

if ((new Process(['git', 'rev-parse', '--is-inside-work-tree']))->run() === 0) {
    (new Process(['git', 'add', $file]))->run();

    $successMessage .= ' and added to git automatically.'
}

$this->io->success($successMessage);
```

Reply:
Resolved in [latest commit](https://github.com/brefphp/bref/pull/1272/commits/ff43ac262ea4429e238bd6f3280109304301e760)